projects/core/src/i18n/translation.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ServerConfig } from '../config/server-config/server-config';
+import { I18NextService } from './i18next/i18next.service';
+
+@Injectable()
+export class TranslationService {
+  private readonly NON_BREAKING_SPACE = String.fromCharCode(160);
+
+  constructor(
+    private i18NextService: I18NextService,
+    private config: ServerConfig
+  ) {}
+
+  exists(key: string, options: any = {}): boolean {
+    return this.i18NextService.exists(key, options);
+  }
+
+  translate(key: string, options: any = {}): string {
+    if (this.i18NextService.exists(key, options)) {
+      return this.i18NextService.t(key, options);
+    } else {
+      this.reportMissingKey(key);
+      return this.getFallbackValue(key);
+    }
+  }
+
+  translateLazy(key: string, options: any = {}): Observable<string> {
+    return new Observable<string>(subscriber => {
+      if (this.i18NextService.exists(key, options)) {
+        subscriber.next(this.i18NextService.t(key, options));
+        subscriber.complete();
+      } else {
+        this.loadKeyNamespace(key, () => {
```

Comment:
Just a question: do we want to display fallback (nbsp) while translations are loading?

Reply:
good point